app/src/js/controllers/ImageController.js

```diff
@@ -47,6 +48,44 @@ angular.module('media_manager')
       CourseCache.current_image = CourseCache.images[ic.index];
       resetBreadcrumb();
     }
-  }
+  };
+
+  ic.newLabel = '';
+  ic.newValue = '';
+  ic.saveMetadata = function(label, value){
+    if(label !== undefined && value !== undefined){
+      ic.image.metadata.forEach(function(item, index, arr){
+        if(item.label == label){
+          arr[index].label = label;
+          arr[index].value = value;
+          Image.update({}, ic.image, function success(data){
+
+          }, function failure(errorResponse){
+            $log.debug("error updating image:", errorResponse);
+          });
+        }
+      });
+    } else {
+      if(ic.newLabel){
+        if(ic.image.metadata == null) {
+          ic.image.metadata = [];
+        }
+        ic.image.metadata.push({'label': ic.newLabel, 'value': ic.newValue});
+        Image.update({}, ic.image, function success(data){
+          ic.newLabel = '';
+          ic.newValue = '';
+          ic.editNewMetadata = false;
+        }, function failure(errorResponse) {
+          $log.debug("error updating image:", errorResponse);
+          ic.image.metadata.pop();
```

Comment:
Same as above about silent failure. We should probably indicate the error in some way on the UI.